tensorflow/python/keras/engine/data_adapter.py

```diff
@@ -1111,6 +1111,7 @@ def __init__(self,
                epochs=1,
                shuffle=False,
                class_weight=None,
+               class_weight_mode=None,
```

Comment:
I would recommend dropping this argument, and simply assuming that the class weights apply to the labels encoded along the last axis.

Reply:
Thanks for picking up this review - looking forward to working with you.

The reason I opted for the argument is that I believe making that assumption would break existing pipelines that are relying on the following current behaviour:
Currently, `fit` will interpret a final label dimension of size 1 or N_classes [if the dimension is <=2]. Your suggestion locks that down to requiring the final dim to be of size N_classes if class weights are being used, thus invalidating the former behaviour, right?

Assuming that the labels must be one-hot encoded (the N_classes-final-dim suggestion) does however resolve another problem, where it'd be ambiguous if final-dim size == N_classes, by coincidence. So it seems the sensible way to go to pick one.

This is all fine with me if it doesn't go against any design principles. I will make the change, just want to make the implications clear!

Reply:
Another consideration: enforcing one-hot representation of labels means that people who depend on running class weights with memory-light, dense label representations will now have to use label tensors that are a factor 'n_classes' larger. Is that acceptable, and am I interpreting your change request correctly?

If we do not assume one-hot representation and do not ask for an argument flag, I think we are back to square 1 on why this feature was never implemented: it's ambiguous to coerce the label format, from the shape of the label alone. This would specifically fail in the case where the final dim size == n_classes, by coincidence. That'd be interpreted as 1-hot when the label was actually dense.